### PR TITLE
perf(frontend): make QueryRunner methods concurrent

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

#14067 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/scheduler/distributed/query.rs [hunk 1]
```diff
@@ -415,12 +417,14 @@ impl QueryRunner {
...
-            if !self.stage_executions[child].is_scheduled().await {
-                return false;
-            }
-        }
-        true
+        self.query
+            .stage_graph
+            .get_child_stages_unchecked(stage_id)
+            .iter()
+            .map(|child| self.stage_executions[child].is_scheduled())
```
Comment (xxchan) [inline src/frontend/src/scheduler/distributed/query.rs:424]: `is_scheduled` only acquires a tokio RwLock. Does it have any improvement to make it concurrent?

File: src/frontend/src/scheduler/distributed/query.rs [hunk 2]
```diff
@@ -415,12 +417,14 @@ impl QueryRunner {
...
+            .collect::<FuturesUnordered<_>>()
+            .all(future::ready)
```
Comment (xxchan) [inline src/frontend/src/scheduler/distributed/query.rs:426]: Although correct, this reads a little weird (all futures is ready..? 🤣).

File: src/frontend/src/scheduler/distributed/query.rs (hunk 1 above)
Comment (liurenjie1024) [reply src/frontend/src/scheduler/distributed/query.rs:424]: +1, I think this doesn't help much here.

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Pull request closed without merging**